src/data/fintype/basic.lean

```diff
@@ -105,6 +105,11 @@ begin
   simp,
 end
 
+@[simp] lemma univ_filter_mem_range (f : α → β) [fintype β]
+  {_ : decidable_pred (λ (b : β), b ∈ set.range f)} [decidable_eq β]:
+  finset.univ.filter (λ y, y ∈ set.range f) = finset.univ.image f :=
```

Comment:
The linter would prefer I write this as something like
```suggestion
  finset.univ.filter (λ y, ∃ x, f x = y) = finset.univ.image f :=
```
but I don't know what what to call the lemma in that form

Reply:
Maybe this?
```lean
@[simp] lemma filter_exists_preimage_eq (f : α → β) [fintype β] [decidable_eq β] :
  univ.filter (λ y, ∃ x, f x = y) = univ.image f :=
by { ext, simp }
```
Or `filter_exists_preimage_eq_image` or `filter_exists_nonempty_fiber_eq`?

Reply:
Just for completeness, another option would be `finset.univ.filter (set.range f) = finset.univ.image f`.

Reply:
It's worth noting that the form that I actually care about in #5269 starts as `finset.univ.filter (λ y, y ∈ f.range)` where `f` is a monoid_hom. Your lemma looks like a nice addition @gebner, but I suspect it will be annoying to convert what I have into it.

Reply:
Ah yes, you're right.  Another option would be to add the following as simp-lemmas, which would be a bit more general-purpose than `filter_exists_preimage_eq`:
 - `filter_exists : s.filter (∃ x, p x) = ⨆ x, s.filter p`
 - `univ_filter_eq : finset.univ.filter (λ y, x = y) = {x}`  (NB: simp can already prove this)
 - `supr_singleton : (⨆ x, {f x}) = finset.image f`

Reply:
Isn't your `filter_exists` missing a lambda binder?

Reply:
You're again right.  I don't know what I was thinking, `finset` doesn't have `⨆` either of course.  But if you use `finset.univ.bind` instead then all the lemmas should already be there except for `filter_exists`.

Reply:
I avoided the word `preimage` because we have `finset.preimage` but this is not a lemma about it.